src/Microsoft.Extensions.Primitives/ChangeToken.cs

```diff
@@ -36,7 +36,12 @@ public static IDisposable OnChange(Func<IChangeToken> changeTokenProducer, Actio
                 // If the token changes after we take the token, then we'll process the update immediately upon
                 // registering the callback.
                 var t = changeTokenProducer();
-                changeTokenConsumer();
+                try
+                {
+                    changeTokenConsumer();
+                }
+                catch // Uncaught exceptions are really bad here
+                { }
```

Comment:
Well, we will have the LoadException event soon which would still trap that, but I can switch this to try/finally if you want, I don't see any downside to try/finally other than we will have a bunch of uncaught exceptions